pkg/controllers/vdb/restart_reconciler.go

```diff
@@ -126,9 +126,14 @@ func (r *RestartReconciler) Reconcile(ctx context.Context, _ *ctrl.Request) (ctr
 	// admin commands to run.  Cluster operations only apply if the entire
 	// vertica cluster is managed by k8s.  We skip that if initPolicy is
 	// ScheduleOnly.
-	if r.PFacts.getUpNodeAndNotReadOnlyCount() == 0 &&
+	if r.PFacts.GetUpNodeAndNotReadOnlyCount() == 0 &&
 		r.Vdb.Spec.InitPolicy != vapi.CommunalInitPolicyScheduleOnly {
-		return r.reconcileCluster(ctx)
+		// Abort cluster restart if at least one pod must not be
+		// restarted.
+		if r.PFacts.GetShutdownCount() == 0 {
```

Comment:
Can we add one more check here? if r.PFacts.GetShutdownCount() == 0 && r.PFacts.GetSandboxName() == vapi.MainCluster

Reply:
Later we will support `stop subcluster` on any sandbox including the main cluster, that is why I did not add an extra check for the sandbox.

Reply:
It is. This path is for start_db. We do not allow start_db if there is at least one node that belongs to a subcluster with shutdown true. So, the moment we find one, we exit. All subclusters must have shutdown false in order to do start_db. I will add a log msg.